lib/app/features/chat/messages/views/components/message_types/audio_message/audio_message.dart

```diff
@@ -0,0 +1,143 @@
+import 'package:ion/app/features/chat/messages/views/components/mesage_timestamp/mesage_timestamp.dart';
+import 'package:ion/app/features/chat/messages/views/components/message_item_wrapper/message_item_wrapper.dart';
+import 'package:ion/app/services/logger/logger.dart';
+import 'package:ion/app/utils/date.dart';
+import 'package:ion/generated/assets.gen.dart';
+import 'package:path_provider/path_provider.dart';
+import 'package:visibility_detector/visibility_detector.dart';
+
+part 'components/audio_wave_form_display.dart';
+part 'components/play_pause_button.dart';
+
+class AudioMessage extends HookWidget {
+  const AudioMessage(
+    this.id, {
+    required this.audioUrl,
+    required this.isMe,
+    super.key,
+  });
+
+  final bool isMe;
+  final String audioUrl;
+  final String id;
+
+  @override
+  Widget build(BuildContext context) {
+    useAutomaticKeepAlive();
+    final audioPlaybackState = useState<PlayerState?>(null);
+    final audioPlaybackController = useAudioPlaybackController();
+
+    final playerWaveStyle = useMemoized(
+      () => PlayerWaveStyle(
+        liveWaveColor:
+            isMe ? context.theme.appColors.onPrimaryAccent : context.theme.appColors.primaryText,
+        fixedWaveColor: context.theme.appColors.sheetLine,
+        seekLineColor: Colors.transparent,
+        waveThickness: 1.0.s,
+        spacing: 2.0.s,
+      ),
+      [isMe],
+    );
+
+    useEffect(
+      () {
+        _initializePlayer(
+          audioPlaybackController,
+          id,
+          audioUrl,
+          playerWaveStyle,
+          audioPlaybackState,
+        );
+
+        // Listen to player state changes
+        audioPlaybackController.onPlayerStateChanged.listen((event) {
+          if (event != PlayerState.stopped) {
+            audioPlaybackState.value = event;
+          }
+        });
+
+        return null;
+      },
+      [],
+    );
+
+    return MessageItemWrapper(
+      isMe: isMe,
+      contentPadding: EdgeInsets.all(12.0.s),
+      child: VisibilityDetector(
+        key: ValueKey(audioUrl),
+        onVisibilityChanged: (info) {
+          if (info.visibleFraction == 0) {
+            audioPlaybackController.pausePlayer();
+          }
+        },
+        child: Row(
+          crossAxisAlignment: CrossAxisAlignment.end,
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: [
+            _PlayPauseButton(
+              audioPlaybackController: audioPlaybackController,
+              audioPlaybackState: audioPlaybackState,
+            ),
+            SizedBox(width: 8.0.s),
+            _AudioWaveformDisplay(
+              audioPlaybackController: audioPlaybackController,
+              audioPlaybackState: audioPlaybackState,
+              playerWaveStyle: playerWaveStyle,
+              isMe: isMe,
+            ),
+            MessageTimeStamp(isMe: isMe),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+Future<void> _initializePlayer(
```

Comment:
let's create a `AudioService` and move logic related to the audio to that service